docs/documentation.html

```diff
@@ -833,6 +833,41 @@ <h3 id="grammar-syntax-and-semantics-parsing-expression-types">Parsing Expressio
     </div>
   </dd>
 
+  <dt><code>!.</code> (END OF INPUT)</dt>
+
+  <dd>
+    <p>Match END OF INPUT. This <em>Bang Dot</em> sequence will match a completely empty input,
+      or patterns up to the end of input. This is a composite rule;
+      It should NOT be placed inside a character set or quote.
```

Comment:
This line seems obvious to me (we don't say this about any of the other primitives), and as such, it was confusing.  Maybe remove?